jina/peapods/pods/__init__.py

```diff
@@ -250,14 +323,19 @@ def __exit__(self, exc_type, exc_val, exc_tb) -> None:
         self.join()
 
     def join(self):
-        """Wait until all peas exit"""
+        """
+        # noqa: DAR201
+        Wait until all peas exit"""
         try:
             for p in self.peas:
                 p.join()
+            for p in self.replica_list:
```

Comment:
I do not think Replicas should be something to be added inside a Pod. A Pod should be considered a replica, and therefore add a construction "around" Pods to handle Pods which are replicas of the same concept.

If this new construction follows the same interface as Pods, the changes are almost transparent to the Flow

Reply:
So inside a Pod, one should still have `peas_args` as `shards` and that should not change imho

Reply:
Also, keeping this hidden, we can profit the constructs of how Flow connects different Pods and constructs the Flow pipeline